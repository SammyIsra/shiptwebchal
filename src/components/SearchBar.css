.SearchBar input {
  border: none;
  border-bottom: 1px solid gray;
  border-radius: 0;
  width: 80%;
}

.SearchBar form {
  width: 100%;
  margin-top: 20px;
  margin-bottom: 20px;
  font-size: 3rem;
}

.SearchBar form button {
  max-width: 20%;
  border: none;
}

.SearchBar input:focus {
  border: gray;
  border-bottom: 1px solid black;
  border-radius: 0;
}

@media only screen and (min-width: 768px){
  .SearchBar form {
    width: 70%;
  }
}