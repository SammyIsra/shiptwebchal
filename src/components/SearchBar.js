import React from 'react';
import {withRouter} from 'react-router-dom';

import './SearchBar.css';

export class SearchBar extends React.Component {
  
  constructor(props){
    super(props);

    this.state = {
      userSearched: ''
    }

    this.onSearchChange = this.onSearchChange.bind(this);
    this.onSearchSubmit = this.onSearchSubmit.bind(this);
}

  onSearchChange(e){
    this.setState({
      userSearched: e.target.value
    })
  }

  //Go to the navigation page
  onSearchSubmit(e){
    this.props.history.push(`/find/${this.state.userSearched}`);
  }

  render(){
    return (
      <div className="SearchBar">
        <form onSubmit={this.onSearchSubmit}> 
          <input 
            type="text" 
            value={this.state.userSearched}
            placeholder="Search for a GitHub username!"
            onChange={this.onSearchChange} />
          <button type="submit">Search</button>
        </form>
      </div>
    );
  }
}

export default withRouter(SearchBar);