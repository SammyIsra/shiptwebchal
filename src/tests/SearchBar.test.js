import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';

import SearchBar from '../components/SearchBar';

test('updates searchbar when typing', ()=>{

  const searchBar = renderer.create(
    <MemoryRouter>
      <SearchBar />
    </MemoryRouter>
  );

  expect(searchBar.toJSON()).toMatchSnapshot();
})