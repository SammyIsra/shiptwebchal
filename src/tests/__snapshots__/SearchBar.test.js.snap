// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updates searchbar when typing 1`] = `
<div
  className="SearchBar"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search for a GitHub username!"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
